mypy/join.py

```diff
@@ -237,7 +249,7 @@ def visit_typeddict_type(self, t: TypedDictType) -> Type:
             required_keys = set(items.keys()) & t.required_keys & self.s.required_keys
             return TypedDictType(items, required_keys, fallback)
         elif isinstance(self.s, Instance):
-            return join_instances(self.s, t.fallback)
+            return join_types(self.s, t.fallback)
```

Comment:
To cover the cases where we join a ``TypedDict`` with a ``Mapping``. If we go directly to join instances, we will not hit the code in ``visit_instance`` that is currently takes care of protocols.